streaming/src/main/scala/io/gearpump/streaming/task/StreamingTransportSerializer.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gearpump.streaming.task
+
+import java.io.DataOutput
+
+import io.gearpump.streaming.{LatencyProbeSerializer, InitialAckRequestSerializer, AckRequestSerializer, AckSerializer}
+import io.gearpump.transport.netty.ITransportMessageSerializer
+import io.gearpump.util.LogUtil
+import org.jboss.netty.buffer.ChannelBuffer
+import org.slf4j.Logger
+
+class StreamingTransportSerializer extends ITransportMessageSerializer{
+  private val log: Logger = LogUtil.getLogger(getClass)
+  private val serializers = new SerializerResolver
+
+  serializers.register(classOf[Ack], new AckSerializer)
+  serializers.register(classOf[AckRequest], new AckRequestSerializer)
+  serializers.register(classOf[InitialAckRequest], new InitialAckRequestSerializer)
+  serializers.register(classOf[LatencyProbe], new LatencyProbeSerializer)
+  serializers.register(classOf[SerializedMessage], new SerializedMessageSerializer)
+
+  override def serialize(dataOutput: DataOutput, obj: Object): Unit = {
+    val registration = serializers.getRegistration(obj.getClass)
+    if(registration != null) {
+      dataOutput.writeInt(registration.id)
```

Comment:
How to ensure that the registration ID is consistent for the whole cluster? I mean, the registration id is generated according to register order. However, how to ensure each node has the same registration order? It is highly possible that some node may have different orders. Maybe we shall put such `SerializerResolver` as part of `AppMaster` so that, for one app, we can have consistent ordering. And of course, as the order is monotonic, you can cache at each worker to speed up.